Source/UnitEx.cpp

```diff
@@ -265,8 +339,12 @@ void UnitEx::DoDock(LOCATION dockLocation)
 	data->numUnits = 1;
 	data->unitId = unitID;
 	data->numWayPoints = 1;
-	data->pts.points.x = dockLocation.x;
-	data->pts.points.y = dockLocation.y;
+	long wpt = 0,
+		x = dockLocation.x,
+		y = dockLocation.y;
+	wpt |= (x & 0x7ff) << 5;
+	wpt |= (y & 0x3ff) << 20;
+	data->pts.rawPoints = wpt;

```

Comment:
At first I thought the indentation here was off, and was tempted to check for tabs/spaces, but it's just a weird multi-variable declaration. I would recommend avoiding multi-variable declarations.

Could simplify initialization of `wpt`:
```cpp
long wpt = ((x & 0x7ff) << 5) | ((y & 0x3ff) << 20);
```
Or just eliminate it entirely:
```cpp
data->pts.rawPoints = ((dockLocation.x & 0x7ff) << 5) | ((dockLocation.y & 0x3ff) << 20);
```

Alternatively, if you wanted to make a `struct Waypoint` datatype, a constructor could handle the above conversion:
```cpp
data->pts.rawPoints = Waypoint(dockLocation.x, dockLocation.y);
```

Reply:
Opened issue #27 to address